compiler/rustc_error_codes/src/error_codes/E0793.md

```diff
@@ -0,0 +1,49 @@
+A lifetime parameter of a function definition is called *late-bound* if it both:
+
+1. appears in an argument type
+2. does not appear in a where clause
```

Comment:
I think it would be better to talk about not appearing in "a generic type constraint" rather than in "a where clause", as the `'a` lifetime parameters in both `fn<'a, T: Trait<'a>>(t: T)` and `fn<'a>(t: impl Trait<'a>)` are early bound without there being any `where` clause.

Reply:
Good point!